File: frontend/src/types/health.ts
/**
 * Health contracts, mirroring `app/schemas/health.py`.
 */

/** Response of `GET /health`. */
export interface HealthResponse {
  /** Always "ok" when the process is serving. */
  status: string;
  service: string;
  version: string;
  environment: string;
}

/** One dependency's verdict. An unavailable optional dependency is not fatal. */
export interface DependencyCheck {
  /** "ok" or "unavailable". */
  status: string;
  required: boolean;
}

/**
 * Response of `GET /health/ready`.
 *
 * Returned with 503 when the verdict is "degraded", so the payload is present
 * on both the success and the failure status.
 */
export interface ReadinessResponse {
  /** "ready" or "degraded". */
  status: string;
  dependencies: Record<string, DependencyCheck>;
}

File: backend/app/schemas/health.py
"""Health endpoint schemas."""

from __future__ import annotations

from pydantic import BaseModel, Field


class HealthResponse(BaseModel):
    """Liveness payload."""

    status: str = Field(description="Always 'ok' when the process is serving.")
    service: str
    version: str
    environment: str


class DependencyCheck(BaseModel):
    """Outcome for a single dependency.

    ``required`` says whether an unavailable dependency makes the service
    unready. Redis is optional until a feature actually needs it.
    """

    status: str = Field(description="'ok' or 'unavailable'.")
    required: bool


class ReadinessResponse(BaseModel):
    """Readiness payload: one entry per dependency, plus an overall verdict."""

    status: str = Field(description="'ready' or 'degraded'.")
    dependencies: dict[str, DependencyCheck]
